x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStopTrainedModelDeploymentAction.java

```diff
@@ -235,24 +227,6 @@ private void normalUndeploy(
         super.doExecute(task, request, finalListener);
     }
 
-    void waitForTaskRemoved(
-        String modelId,
-        TrainedModelAllocation trainedModelAllocation,
-        StopTrainedModelDeploymentAction.Request request,
-        StopTrainedModelDeploymentAction.Response response,
-        ActionListener<StopTrainedModelDeploymentAction.Response> listener
-    ) {
-        final Set<String> nodesOfConcern = trainedModelAllocation.getNodeRoutingTable().keySet();
-        client.admin()
-            .cluster()
-            .prepareListTasks(nodesOfConcern.toArray(String[]::new))
-            .setDetailed(true)
-            .setWaitForCompletion(true)
-            .setActions(modelId)
```

Comment:
I think this bug was added in: #81259

The actions used to contain the model id. Regardless, the new stopping path is much cleaner.